ipa-core/src/cli/hpke.rs

```diff
@@ -0,0 +1,337 @@
+    #[arg(long, value_name = "FILE")]
+    output_dir: PathBuf,
+    /// Path to helper network configuration file
+    #[arg(long)]
+    network: PathBuf,
+}
+
+#[derive(Debug, Parser)]
+#[clap(name = "test_decrypt", about = "Test Decrypt")]
+#[command(about)]
+pub struct DecryptArgs {
+    /// Path to helper1 file to decrypt
+    #[arg(long)]
+    input_file1: PathBuf,
+
+    /// Helper1 Private key for decrypting match keys
+    #[arg(long)]
+    mk_private_key1: PathBuf,
+
+    /// Path to helper2 file to decrypt
+    #[arg(long)]
+    input_file2: PathBuf,
+
+    /// Helper2 Private key for decrypting match keys
+    #[arg(long)]
+    mk_private_key2: PathBuf,
+
+    /// Path to helper3 file to decrypt
+    #[arg(long)]
+    input_file3: PathBuf,
+
+    /// Helper3 Private key for decrypting match keys
+    #[arg(long)]
+    mk_private_key3: PathBuf,
+
+    /// The destination file for decrypted output.
+    #[arg(long, value_name = "FILE")]
+    output_file: PathBuf,
+}
+
+/// # Panics
+/// if input file or network file are not correctly formatted
+/// # Errors
+/// if it cannot open the files
+pub fn encrypt(args: &EncryptArgs) -> Result<(), BoxError> {
+    let input = InputSource::from_file(&args.input_file);
+    let records = input.iter::<TestRawDataRecord>().collect::<Vec<_>>();
+
+    let mut rng = StdRng::from_entropy();
+    let mut key_registries = KeyRegistries::default();
+    let network = NetworkConfig::from_toml_str(&read_to_string(&args.network).unwrap()).unwrap();
+    let Some((key_id, key_registries)) = key_registries.init_from(&network, DEFAULT_KEY_ID) else {
+        panic!("could not load network file")
+    };
+    let shares: [Vec<OprfReport<BreakdownKey, TriggerValue, Timestamp>>; 3] =
+        records.iter().cloned().share();
+
+    let shares_key_registries = zip(shares, key_registries);
+
+    for (index, (shares, key_registry)) in shares_key_registries.enumerate() {
+        let mut writer: Box<dyn Write> = Box::new(
+            OpenOptions::new()
+                .write(true)
+                .create_new(true)
+                .open(args.output_dir.join(format!("helper{}.enc", index + 1)))?,
+        );
+        for share in shares {
+            let output = share.encrypt(key_id, key_registry, &mut rng).unwrap();
+            let hex_output = hex::encode(&output);
+            writeln!(writer, "{hex_output}")?;
+        }
+    }
+
+    Ok(())
+}
+
+async fn build_hpke_registry(
+    private_key_file: PathBuf,
+) -> Result<KeyRegistry<PrivateKeyOnly>, BoxError> {
+    let mk_encryption = Some(HpkeServerConfig::File { private_key_file });
+    let key_registry = hpke_registry(mk_encryption.as_ref()).await?;
+    Ok(key_registry)
+}
+
+/// # Panics
+// if input files or private_keys are not correctly formatted
+/// # Errors
+/// if it cannot open the files
+pub async fn decrypt_and_reconstruct(args: DecryptArgs) -> Result<(), BoxError> {
+    let file1 = File::open(args.input_file1)?;
```

Comment:
It is probably better to create a routine that opens a file and produces a stream/iterator over encrypted reports. Then, the main routine can just zip them together and reconstruct reports